[jit] Put all modules in the global Python CU

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23154 [jit] Put all modules in the global Python CU**

The last of the functions/classes/modules trio that end up in
CompilationUnits.

I'm soliciting ideas for how to mangle the names of modules. Since
modules are singleton classes, each instance of a ScriptModule has its
own ClassType and they all have the same qualified name, so we need to
mangle them somehow.

If we mangle them the same way we mangle functions (append a
`__torch_mangle_0` at the end), then all `check_trace` and expect files
fail because we're littering the mangled name everywhere.

So the current approach is to mangle the namespace itself: e.g.
`foo.bar.MyMod` => `foo.bar.__torch_mangle_9.MyMod`. I'm not happy with
this though so I hope there's a better idea.

gh-metadata: pytorch pytorch 23154 gh/suo/102/head


Differential Revision: [D16441913](https://our.internmc.facebook.com/intern/diff/D16441913)

suo committed:
[jit] Put all modules in the global Python CU

The last of the functions/classes/modules trio that end up in
CompilationUnits.

I'm soliciting ideas for how to mangle the names of modules. Since
modules are singleton classes, each instance of a ScriptModule has its
own ClassType and they all have the same qualified name, so we need to
mangle them somehow.

If we mangle them the same way we mangle functions (append a
`__torch_mangle_0` at the end), then all `check_trace` and expect files
fail because we're littering the mangled name everywhere.

So the current approach is to mangle the namespace itself: e.g.
`foo.bar.MyMod` => `foo.bar.__torch_mangle_9.MyMod`. I'm not happy with
this though so I hope there's a better idea.

suo committed:
Update on "[jit] Put all modules in the global Python CU"

[jit] Put all modules in the global Python CU

The last of the functions/classes/modules trio that end up in
CompilationUnits.

I'm soliciting ideas for how to mangle the names of modules. Since
modules are singleton classes, each instance of a ScriptModule has its
own ClassType and they all have the same qualified name, so we need to
mangle them somehow.

If we mangle them the same way we mangle functions (append a
`__torch_mangle_0` at the end), then all `check_trace` and expect files
fail because we're littering the mangled name everywhere.

So the current approach is to mangle the namespace itself: e.g.
`foo.bar.MyMod` => `foo.bar.__torch_mangle_9.MyMod`. I'm not happy with
this though so I hope there's a better idea.

gh-metadata: pytorch pytorch 23154 gh/suo/102/head

suo committed:
Update on "[jit] Put all modules in the global Python CU"

[jit] Put all modules in the global Python CU

The last of the functions/classes/modules trio that end up in
CompilationUnits.

I'm soliciting ideas for how to mangle the names of modules. Since
modules are singleton classes, each instance of a ScriptModule has its
own ClassType and they all have the same qualified name, so we need to
mangle them somehow.

If we mangle them the same way we mangle functions (append a
`__torch_mangle_0` at the end), then all `check_trace` and expect files
fail because we're littering the mangled name everywhere.

So the current approach is to mangle the namespace itself: e.g.
`foo.bar.MyMod` => `foo.bar.__torch_mangle_9.MyMod`. I'm not happy with
this though so I hope there's a better idea.

gh-metadata: pytorch pytorch 23154 gh/suo/102/head

suo committed:
Update on "[jit] Put all modules in the global Python CU"

[jit] Put all modules in the global Python CU

The last of the functions/classes/modules trio that end up in
CompilationUnits.

I'm soliciting ideas for how to mangle the names of modules. Since
modules are singleton classes, each instance of a ScriptModule has its
own ClassType and they all have the same qualified name, so we need to
mangle them somehow.

If we mangle them the same way we mangle functions (append a
`__torch_mangle_0` at the end), then all `check_trace` and expect files
fail because we're littering the mangled name everywhere.

So the current approach is to mangle the namespace itself: e.g.
`foo.bar.MyMod` => `foo.bar.__torch_mangle_9.MyMod`. I'm not happy with
this though so I hope there's a better idea.

gh-metadata: pytorch pytorch 23154 gh/suo/102/head

zdevito reviewed (approved):
This is ok, but shouldn't we be consistent with mangling? I am concerned about doing one thing for functions and another for Modules. Wouldn't the mangling show up in stack traces for Functions? Isn't that a similar concern?

suo commented:
yeah, I'm going to change functions to be the same way. Right now it didn't show up because we don't leave function calls in the graph ever.

suo committed:
Update on "[jit] Put all modules in the global Python CU"

[jit] Put all modules in the global Python CU

The last of the functions/classes/modules trio that end up in
CompilationUnits.

I'm soliciting ideas for how to mangle the names of modules. Since
modules are singleton classes, each instance of a ScriptModule has its
own ClassType and they all have the same qualified name, so we need to
mangle them somehow.

If we mangle them the same way we mangle functions (append a
`__torch_mangle_0` at the end), then all `check_trace` and expect files
fail because we're littering the mangled name everywhere.

So the current approach is to mangle the namespace itself: e.g.
`foo.bar.MyMod` => `foo.bar.__torch_mangle_9.MyMod`. I'm not happy with
this though so I hope there's a better idea.

gh-metadata: pytorch pytorch 23154 gh/suo/102/head

suo committed:
Update on "[jit] Put all modules in the global Python CU"

[jit] Put all modules in the global Python CU

The last of the functions/classes/modules trio that end up in
CompilationUnits.

I'm soliciting ideas for how to mangle the names of modules. Since
modules are singleton classes, each instance of a ScriptModule has its
own ClassType and they all have the same qualified name, so we need to
mangle them somehow.

If we mangle them the same way we mangle functions (append a
`__torch_mangle_0` at the end), then all `check_trace` and expect files
fail because we're littering the mangled name everywhere.

So the current approach is to mangle the namespace itself: e.g.
`foo.bar.MyMod` => `foo.bar.__torch_mangle_9.MyMod`. I'm not happy with
this though so I hope there's a better idea.

gh-metadata: pytorch pytorch 23154 gh/suo/102/head

suo committed:
Update on "[jit] Put all modules in the global Python CU"


The last of the functions/classes/modules trio that end up in
CompilationUnits.

I'm soliciting ideas for how to mangle the names of modules. Since
modules are singleton classes, each instance of a ScriptModule has its
own ClassType and they all have the same qualified name, so we need to
mangle them somehow.

If we mangle them the same way we mangle functions (append a
`__torch_mangle_0` at the end), then all `check_trace` and expect files
fail because we're littering the mangled name everywhere.

So the current approach is to mangle the namespace itself: e.g.
`foo.bar.MyMod` => `foo.bar.__torch_mangle_9.MyMod`. I'm not happy with
this though so I hope there's a better idea.

gh-metadata: pytorch pytorch 23154 gh/suo/102/head

Differential Revision: [D16441913](https://our.internmc.facebook.com/intern/diff/D16441913)

suo committed:
Update on "[jit] Put all modules in the global Python CU"


The last of the functions/classes/modules trio that end up in
CompilationUnits.

I'm soliciting ideas for how to mangle the names of modules. Since
modules are singleton classes, each instance of a ScriptModule has its
own ClassType and they all have the same qualified name, so we need to
mangle them somehow.

If we mangle them the same way we mangle functions (append a
`__torch_mangle_0` at the end), then all `check_trace` and expect files
fail because we're littering the mangled name everywhere.

So the current approach is to mangle the namespace itself: e.g.
`foo.bar.MyMod` => `foo.bar.__torch_mangle_9.MyMod`. I'm not happy with
this though so I hope there's a better idea.

gh-metadata: pytorch pytorch 23154 gh/suo/102/head


Differential Revision: [D16441913](https://our.internmc.facebook.com/intern/diff/D16441913)

suo committed:
Update on "[jit] Put all modules in the global Python CU"


The last of the functions/classes/modules trio that end up in
CompilationUnits.

I'm soliciting ideas for how to mangle the names of modules. Since
modules are singleton classes, each instance of a ScriptModule has its
own ClassType and they all have the same qualified name, so we need to
mangle them somehow.

If we mangle them the same way we mangle functions (append a
`__torch_mangle_0` at the end), then all `check_trace` and expect files
fail because we're littering the mangled name everywhere.

So the current approach is to mangle the namespace itself: e.g.
`foo.bar.MyMod` => `foo.bar.__torch_mangle_9.MyMod`. I'm not happy with
this though so I hope there's a better idea.

gh-metadata: pytorch pytorch 23154 gh/suo/102/head


Differential Revision: [D16441913](https://our.internmc.facebook.com/intern/diff/D16441913)

suo committed:
Update on "[jit] Put all modules in the global Python CU"


The last of the functions/classes/modules trio that end up in
CompilationUnits.

I'm soliciting ideas for how to mangle the names of modules. Since
modules are singleton classes, each instance of a ScriptModule has its
own ClassType and they all have the same qualified name, so we need to
mangle them somehow.

If we mangle them the same way we mangle functions (append a
`__torch_mangle_0` at the end), then all `check_trace` and expect files
fail because we're littering the mangled name everywhere.

So the current approach is to mangle the namespace itself: e.g.
`foo.bar.MyMod` => `foo.bar.__torch_mangle_9.MyMod`. I'm not happy with
this though so I hope there's a better idea.

gh-metadata: pytorch pytorch 23154 gh/suo/102/head


Differential Revision: [D16441913](https://our.internmc.facebook.com/intern/diff/D16441913)

suo committed:
Update on "[jit] Put all modules in the global Python CU"


The last of the functions/classes/modules trio that end up in
CompilationUnits.

I'm soliciting ideas for how to mangle the names of modules. Since
modules are singleton classes, each instance of a ScriptModule has its
own ClassType and they all have the same qualified name, so we need to
mangle them somehow.

If we mangle them the same way we mangle functions (append a
`__torch_mangle_0` at the end), then all `check_trace` and expect files
fail because we're littering the mangled name everywhere.

So the current approach is to mangle the namespace itself: e.g.
`foo.bar.MyMod` => `foo.bar.__torch_mangle_9.MyMod`. I'm not happy with
this though so I hope there's a better idea.

gh-metadata: pytorch pytorch 23154 gh/suo/102/head


Differential Revision: [D16441913](https://our.internmc.facebook.com/intern/diff/D16441913)

suo committed:
Update on "[jit] Put all modules in the global Python CU"


The last of the functions/classes/modules trio that end up in
CompilationUnits.

I'm soliciting ideas for how to mangle the names of modules. Since
modules are singleton classes, each instance of a ScriptModule has its
own ClassType and they all have the same qualified name, so we need to
mangle them somehow.

If we mangle them the same way we mangle functions (append a
`__torch_mangle_0` at the end), then all `check_trace` and expect files
fail because we're littering the mangled name everywhere.

So the current approach is to mangle the namespace itself: e.g.
`foo.bar.MyMod` => `foo.bar.__torch_mangle_9.MyMod`. I'm not happy with
this though so I hope there's a better idea.

gh-metadata: pytorch pytorch 23154 gh/suo/102/head


Differential Revision: [D16441913](https://our.internmc.facebook.com/intern/diff/D16441913)
